pkg/proxy/types.go

```diff
@@ -42,3 +43,33 @@ type ServicePortName struct {
 func (spn ServicePortName) String() string {
 	return fmt.Sprintf("%s:%s", spn.NamespacedName.String(), spn.Port)
 }
+
+// ServicePort is an interface which abstracts a service info.
+type ServicePort interface {
+	// String returns service string.
+	String() string
+	// ClusterIP returns service cluster IP.
+	ClusterIP() string
+	// Protocol returns service protocol.
+	Protocol() api.Protocol
+	// HealthCheckNodePort returns service health check node port if present.  If return 0, it means not present.
+	HealthCheckNodePort() int
+}
+
+// Endpoints is an interface which abstracts an endpoints info.
```

Comment:
After taking a second look and considering this will be a "public" API, I think the command should read like "Endpoints in an interface which abstracts information about an endpoint". This seems more generic to me, WDYT?

Reply:
Yeah, it's a good comment message :) Thanks! It's fixed now.